eta-meeting-organizer-backend/src/main/java/hu/flowacademy/meetingorganizer/service/BuildingService.java

```diff
@@ -67,13 +70,18 @@ public void validateBuildingOnCreate(BuildingDTO input) {
     }
   }
 
-  public void validateBuildingOnUpdate(BuildingDTO input) {
+  public void validateBuildingOnUpdate(Long id, BuildingDTO input) {
     validateBuildingData(input);
-    List<Building> result = buildingRepository
-        .findByCityAndBuildingName(input.getCity(), input.getBuildingName());
-    result.remove(input.toEntity());
-    if (!(result.isEmpty())  && (buildingRepository.findAllAddresses()
-        .contains(input.getAddress()))){
+    Building formerBuilding = buildingRepository.findById(id)
+        .orElseThrow(BuildingNotFoundException::new);
+    List<String> addressList = buildingRepository.findAllAddresses();
```

Comment:
Ez lehetne inkább úgy, hogy `findByAddressNotIn(formerBuilding.getAddress())` és akkor nem kell a remove

Reply:
próbáltuk Tomival ezt a queryt, de nekünk nem működött

Reply:
Úgy próbáltátok, hogy a paraméter egy kollekció? Megnéztem most a dokumentációt, talán ez lehet a gond. pl.: `findByAddressNotIn(List.of(formerBuilding.getAddress()))`

Reply:
Kijavítottam.